jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -463,14 +465,51 @@ def mos_meta_parser(app, data_obj, ids=None, spectra=False, sp1d=False):
             _add_to_table(app, dec, "Dec.")
 
 
-def _get_object_keyword(data_obj, ids):
-    if ids is not None:
-        # remove leading path to file name
-        ids_short = [os.path.basename(d) for d in ids]
-        names = [x[0].header.get("OBJECT", d) for x, d in zip(data_obj, ids_short)]
-    else:
-        names = [x[0].header.get("OBJECT", FALLBACK_NAME) for x in data_obj]
-    return names
+def _get_source_names_by_hdu(hdus, filepaths=None, header_keys=['SOURCEID', 'OBJECT']):
+    """
+    Attempts to extract a list of source identifiers via different header values.
+
+    Parameters
+    ----------
+    hdus : list of HDU
+        A list of HDUs (NOT an HDUList!) to check the header of. Filtering should
+        be done in advance to decide which HDUs to check against (e.g., SCI only).
+    filepaths : list of str, optional
+        A list of filepaths to fallback on if no header values are identified.
+    header_keys: str or list of str, optional
+        The header value (or values) to search an HDU for to extract the source id.
+        List order is assumed to be priority order (i.e. will return first successful
+        lookup)
+    """
+    src_names = list()
+    # If the user only provided one key to search, put it in a list for the upcoming for loop
+    if isinstance(header_keys, Iterable) and not isinstance(header_keys, (str, dict)):
```

Comment:
This comment doesn't seem to match the functionality of the code - the comment seems to imply that if it's not iterable it should be put in a list e.g. `[header_keys]`, but it doesn't do anything in that case. Is this doing everything that was intended?

Reply:
You are 100% correct! This is entirely a bug. I've pushed a fix and clarified a comment. This slipped by because I forgot to write a test for it when the feature was suggested, so I wrote a few tests to capture it. (Proud of me Pey Lian? 😄)